src/runtime/cuda.cpp

```diff
@@ -505,6 +470,33 @@ WEAK bool validate_device_pointer(void *user_context, halide_buffer_t *buf, size
 #endif
 }
 
+WEAK CUmodule compile_kernel(void *user_context, const char *ptx_src, int size) {
+    debug(user_context) << "CUDA: compile_kernel cuModuleLoadData " << (void *)ptx_src << ", " << size << " -> ";
+
+    CUjit_option options[] = {CU_JIT_MAX_REGISTERS};
+    unsigned int max_regs_per_thread = 64;
+
+    // A hack to enable control over max register count for
+    // testing. This should be surfaced in the schedule somehow
+    // instead.
+    char *regs = getenv("HL_CUDA_JIT_MAX_REGISTERS");
```

Comment:
Hmm... could someone cause a Halide-generated kernel to crash or misbehave in "real" code by defining this?

Reply:
Nah, it just makes it faster or slower depending on the setting.

Reply:
Oh, but some kernels will refuse to run entirely if you're using a very large thread block and you set this high, so yes, it's possible to get Halide to error out too for some pipelines.

Reply:
(This is old existing code, not new to this branch)

Reply:
Feel free to file an issue if you like, but changing it in this PR is not a good idea.